src/sagas/index.js

```diff
@@ -143,85 +143,115 @@ export function* transformAndPermitVm (vm) {
  */
 export function* fetchByPage () {
   const [
-    vmsPage,
-    vmsExpectMorePages,
-    poolsPage,
-    poolsExpectMorePages,
+    vmsCount, vmsPage, vmsExpectMorePages,
+    poolsCount, poolsPage, poolsExpectMorePages,
   ] = yield select(({ vms }) => [
-    vms.get('vmsPage'), !!vms.get('vmsExpectMorePages'),
-    vms.get('poolsPage'), !!vms.get('poolsExpectMorePages'),
+    vms.get('vms').size, vms.get('vmsPage'), !!vms.get('vmsExpectMorePages'),
+    vms.get('pools').size, vms.get('poolsPage'), !!vms.get('poolsExpectMorePages'),
   ])
 
-  function* currentVmsIds ({ payload: { count, page } }) {
-    const start = count * page
-    const end = start + count
-    return Array.from(yield select(state => state.vms.get('vms').keys())).slice(start, end)
-  }
-
-  function* currentPoolsIds ({ payload: { count, page } }) {
-    const start = count * page
-    const end = start + count
-    return Array.from(yield select(state => state.vms.get('pools').keys())).slice(start, end)
-  }
-
   //
-  // If more pages are expected, fetch the next page and grab the ids fetched
-  // If no more pages are expected, grab the current page of ids from the redux store
+  // If more pages are expected, fetch the next page
+  // If no more pages are expected, skip the fetch
   //
   const count = AppConfiguration.pageLimit
-  const [vms, pools] = yield all([
-    call(vmsExpectMorePages ? fetchVms : currentVmsIds, { payload: { count, page: vmsPage + 1 } }),
-    call(poolsExpectMorePages ? fetchPools : currentPoolsIds, { payload: { count, page: poolsPage + 1 } }),
-  ])
+  const {
+    vms: { internalVms: vms = null },
```

Comment:
`null` breaks default values - empty array or `undefined` seems safer

Reply:
Would this look better?
```jsx
  const [
    { internalVms: vms = null },
    { internalPools: pools = null },
  ] = yield all([ 
    ...
  ])
```

Really I want `vms` and `pools` to be `null` if the fetch is skipped.  I could change it to `false` but I think that would make it worse.  Empty array isn't viable for a "nothing to do" value since empty array is a valid value.

Reply:
Sorry for not being precise about that - I was concerned with `null` value breaking nested destructuring which depends on default values in order to be safe. The same applies to other places were default values are used. Since we do depend on them in many places I would rather limit the use of `null` in our code. In this case empty array seems OK (the code should support this case anyway). Using `undefined` also seems possible.

Reply:
Note: I took the default value out and just moved the `null` to the no fetch value itself.  Looks/feels a bit cleaner that way.